### DISCUSSION TITLE
Should chains add rpc/explorers url available test check?

### DISCUSSION THREAD

**Discussion by izayl** [wevm/viem#1303] [Idea / Feature Request]
As anyone can create PR to add a chain to viem, but there is no test to check weather the urls is available or not

I just wrote a rpc check, and I found the exist chains rpcUrls  are not 100% avilable

```ts
test.each(Object.values(chains))('chain $name http rpc', async (chain) => {
  const rpcUrls = [...chain.rpcUrls.default.http, ...chain.rpcUrls.public.http]

  for (const url of rpcUrls) {
    await expect(
      rpc.http(url, { body: { method: 'eth_chainId' } }).then((r) => r.result),
    ).resolves.toEqual(`0x${BigInt(chain.id).toString(16)}`)
  }
})

```

the result
```sh
Tests  31 failed | 114 passed
```

should add test to check the urls in chain defines?

**Comment by tmm** [comment]
Interesting idea! This should at least be a script in the codebase.

Adding this to the test suite might slow things down though as public resources, especially for less popular chains, tend to go down or get rate-limited. In these situations, it might be a pain to find a new RPC URL or wait for the rate limit to lift just to make the tests pass.

**Comment by izayl** [comment]
ok, let me create a PR to add a script for chains urls check

**Comment by izayl** [comment]
https://github.com/wagmi-dev/viem/pull/1309
